### PR TITLE
change the warnings in cqr predictions

### PR DESCRIPTION
**Pull request by tmorzade**
# Description

Fixes a bug in two warnings in the MapieQuantileRegressor predictions

## Type of change

Please remove options that are irrelevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `make lint`
- [x] Typing passes successfully : `make type-check`
- [x] Unit tests pass successfully : `make tests`
- [x] Coverage is 100% : `make coverage`
- [x] Documentation builds successfully : `make doc`

### CODE REVIEW

File: mapie/utils.py [hunk 1]
```diff
@@ -414,73 +415,83 @@ def check_lower_upper_bounds(
...
+    )
+
+    if any_final_inversion:
         warnings.warn(
-            "WARNING: Following the additional value added to have conformal "
-            "predictions, the upper and lower bound present issues as one "
-            "might be higher or lower than the other."
+            "WARNING: The predictions have issues.\n"
+            + "The upper predictions are lower than"
+            + "the lower predictions at some points."
```
Comment (vincentblot28) [inline mapie/utils.py:494]: Upper quantile and lower quantile ?

**Review by LacombeLouis — APPROVED** [review]
This corrects the main issues pointed out: warning phrasing and the order of the element in lign 459. However, it does not fully satisfy some of the comments from @gmartinonQM with regard to a more explicit warning message.

File: mapie/quantile_regression.py [hunk 2]
```diff
@@ -656,9 +656,9 @@ def predict(
         n = self.n_calib_samples
         q = (1 - (alpha)) * (1 + (1 / n))
 
-        y_preds = np.full(
+        y_preds = np.empty(
```
Comment (LacombeLouis) [inline mapie/quantile_regression.py:659]: Not sure why we are making this change? Why do we want to set the values to 0 as oppose to `np.nan`?

File: mapie/tests/test_utils.py [hunk 3]
```diff
@@ -71,8 +79,8 @@ def test_check_null_weight_with_zeros() -> None:
 @pytest.mark.parametrize("estimator", [LinearRegression(), DumbEstimator()])
 @pytest.mark.parametrize("sample_weight", [None, np.ones_like(y_toy)])
 def test_fit_estimator(
-    estimator: Any, sample_weight: Optional[ArrayLike]
-) -> None:
+        estimator: Any,
+        sample_weight: Optional[ArrayLike]) -> None:
```
Comment (LacombeLouis) [inline mapie/tests/test_utils.py:83]: I believe this is the correct indentation?

def test_fit_estimator(
        estimator: Any,
        sample_weight: Optional[ArrayLike]
) -> None:

File: mapie/utils.py [hunk 4]
```diff
@@ -271,15 +271,16 @@ def check_n_features_in(
...
         if cast(Any, estimator).n_features_in_ != n_features_in:
             raise ValueError(
-                "Invalid mismatch between "
+                "Invalid mismatch between ",
                 "X.shape and estimator.n_features_in_."
             )
     return n_features_in
 
 
 def check_alpha_and_n_samples(
```
Comment (LacombeLouis) [inline mapie/utils.py:280]: I believe this is the correct indetation?

def check_alpha_and_n_samples(
        alphas: Union[Iterable[float], float],
        n: int
) -> None:

**Review by LacombeLouis — CHANGES_REQUESTED** [review]

**Review by LacombeLouis — APPROVED** [review]

**Pull request merged**